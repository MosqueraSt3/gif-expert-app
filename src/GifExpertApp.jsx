import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    const [ categories, setCategories ] = useState(['One punch man']);

    const onAddCategory = (inputValue) => {
        if(categories.find((category) => category === inputValue)) return;
        setCategories([ inputValue, ...categories ]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                onAddCategory={ onAddCategory }

            />

            { 
                categories.map((category) => (
                    <GifGrid key={ category } category={ category } />
                )) 
            }
        </>
    )
}
