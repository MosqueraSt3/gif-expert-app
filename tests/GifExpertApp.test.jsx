import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('<GifExpertApp /> component', () => { 

    test('should match with the snapshot', () => { 
        const { container } = render(<GifExpertApp />);
        expect( container ).toMatchSnapshot();
    })

    test('Should containt the title', () => {

        render( <GifExpertApp /> );

        expect(screen.getByText('GifExpertApp'));

    })

})