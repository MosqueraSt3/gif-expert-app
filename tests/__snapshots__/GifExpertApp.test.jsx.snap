// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> component should match with the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One punch man
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
